src/test/java/bio/terra/service/snapshot/SnapshotExportIntegrationTest.java

```diff
@@ -234,19 +259,46 @@ public void snapshotExportTest() throws Exception {
             .getService();
 
     for (var path : paths) {
-      Blob blob = authedStorage.get(GcsUriUtils.parseBlobUri(parquet.getManifest()));
-      assertThat("Authorized user can read " + path, blob, notNullValue());
+      if (signUrls) {
+        // Make sure we can download the parquet bytes
+        byte[] bytes = readUrl(path);
+        assertThat(
+            "Signed URL is accessible and parquet isn't empty", bytes.length, greaterThan(0));
+      } else {
+        Blob blob = authedStorage.get(GcsUriUtils.parseBlobUri(parquet.getManifest()));
+        assertThat("Authorized user can read " + path, blob, notNullValue());
+
+        StorageException notAuthorizedException = null;
+        try {
+          unauthedStorage.get(GcsUriUtils.parseBlobUri(parquet.getManifest()));
+        } catch (StorageException ex) {
+          notAuthorizedException = ex;
+        }
+        // Appeasing SpotBugs
+        assert notAuthorizedException != null;
+        assertThat(
+            "Unauthorized user cannot read " + path,
+            notAuthorizedException.getCode(),
+            equalTo(403));
+      }
+    }
+  }
 
-      StorageException notAuthorizedException = null;
-      try {
-        unauthedStorage.get(GcsUriUtils.parseBlobUri(parquet.getManifest()));
-      } catch (StorageException ex) {
-        notAuthorizedException = ex;
+  private byte[] readUrl(String url) throws IOException {
+    try (CloseableHttpClient client = HttpClients.createDefault()) {
+      HttpUriRequest request = new HttpGet(url);
+      try (CloseableHttpResponse response = client.execute(request); ) {
+        assertThat(
+            "Signed URL is accessible",
+            response.getStatusLine().getStatusCode(),
+            equalTo(HttpStatus.OK.value()));
+
+        byte[] bytes;
+        try (InputStream responseBody = response.getEntity().getContent()) {
```

Comment:
Is there a reason to do it this way? You can read directly from a `URL` using `URL.openStream()`.
```
  private byte[] readUrl(String url) throws IOException {
    try (var responseBody = new URL(url).openStream()) {
      return responseBody.readAllBytes();
    }
  }
```

Reply:
Oh interesting. I was doing this the way that we've done it in other places where we do a `HEAD` on the file.  Lemme use this instead

Reply:
Actually, I'm going to leave as is since my way explicitly checks for a successful status (whereas presumably I could end up having a 400 or 403 back here and this would just return the error body)

Reply:
You can try and see, but I'm pretty sure a 400 or 403 would result in a thrown `IOException`. Looking at the source, any status >=400 results in an `IOException`. https://github.com/openjdk/jdk17/blob/master/src/java.base/share/classes/sun/net/www/protocol/http/HttpURLConnection.java#L1991C3-L1991C3

Reply:
Ah fair enough.  I'll use this then!